registry/client/src/sharedLibs/Edit.js

```diff
@@ -0,0 +1,92 @@
+import React, {Fragment} from 'react';
+import {
+    Create,
+    Edit,
+    FormTab,
+    TabbedForm,
+    TextInput,
+    FormDataConsumer,
+    TextField,
+} from 'react-admin'; // eslint-disable-line import/no-unresolved
+
+import * as validators from '../validators';
+import Title from './Title';
+
+const selectHasAssetsDiscoveryUrl = (formData) => formData.assetsDiscoveryUrl && formData.assetsDiscoveryUrl.length !== 0;
+const selectWarnMessageDueToAssetsDiscoveryUrl = (formData) => {
+    if (!selectHasAssetsDiscoveryUrl(formData)) {
+        return '';
+    }
+
+    return `Do not need to specify SPA bundle because they would be fetched and set from assets discovery URL if they exist there`;
+};
+
+/**
+ * Need to validate 'assetsDiscoveryUrl', 'spaBundle' here because they both depend on each other
+ */
+const validateSharedLib = (values) => {
+    const errors = {};
+
+    if (!values.name) {
+        errors.name = validators.required(values.name);
+    } else {
+        errors.name = validators.disallowedWhiteSpaces(values.name);
+    }
+
+    if (values.assetsDiscoveryUrl) {
+        errors.assetsDiscoveryUrl = validators.url(values.assetsDiscoveryUrl);
+    } else {
+        errors.spaBundle = values.spaBundle ? validators.url(values.spaBundle) : validators.required(values.spaBundle);
+    }
+
+    return errors;
+};
+
+const InputForm = ({mode = 'edit', ...props}) => {
+    return (
+        <TabbedForm {...props}>
+            <FormTab label="Summary">
+                {mode === 'edit'
+                    ? <TextField source="name" />
+                    : <TextInput source="name" fullWidth />}
```

Comment:
I think we shouldn't allow change of name since it's a primary key and ID of the entity

Reply:
but here we use TextFIELD for "edit" and "TextINPUT" for "view",
or I missed something?